### PR TITLE
Add dockerfile and CI

### PR DESCRIPTION
**Pull request by alexhumphreys**
Starting a new repo just for this to make it easier to publish publicly on ghcr.io.

### CODE REVIEW

File: .github/workflows/build_network_tools.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v2.4.0
+
+      - name: Environment standardization
+        run: |
+          default_branch=$(git remote show origin | grep 'HEAD branch' | cut -d ' ' -f 5)
+          echo "default_branch=${default_branch}" >> $GITHUB_ENV
+          default_branch_ref="refs/heads/${default_branch}"
+          echo "default_branch_ref=${default_branch_ref}" >> $GITHUB_ENV
+          repo="${{ github.event.repository.name }}"
```
Comment (dorsradek) [inline .github/workflows/build_network_tools.yaml:28]: I think this one is not used

File: .github/workflows/build_network_tools.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2.4.0
+
+      - name: Environment standardization
+        run: |
+          default_branch=$(git remote show origin | grep 'HEAD branch' | cut -d ' ' -f 5)
+          echo "default_branch=${default_branch}" >> $GITHUB_ENV
+          default_branch_ref="refs/heads/${default_branch}"
+          echo "default_branch_ref=${default_branch_ref}" >> $GITHUB_ENV
```
Comment (dorsradek) [inline .github/workflows/build_network_tools.yaml:27]: we could just inline this, but up to you

File: .github/workflows/build_network_tools.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+          IMAGE_ID=ghcr.io/${{ github.repository_owner }}/$IMAGE_NAME
+
+          # Change all uppercase to lowercase
+          IMAGE_ID=$(echo $IMAGE_ID | tr '[A-Z]' '[a-z]')
+          # Strip git ref prefix from version
+          VERSION=${{ github.sha }}
+          echo IMAGE_ID=$IMAGE_ID
+          echo VERSION=$VERSION
+          docker tag ${{ github.sha }} $IMAGE_ID:$VERSION
+          docker tag ${{ github.sha }} $IMAGE_ID:latest
```
Comment (dorsradek) [inline .github/workflows/build_network_tools.yaml:62]: do we want to push latest?

Comment (dorsradek) [reply .github/workflows/build_network_tools.yaml:62]: no strong opinion here. I am ok with both

**Review by dorsradek — APPROVED** [review]

File: .github/workflows/build_network_tools.yaml (hunk 1 above)
Comment (alexhumphreys) [reply .github/workflows/build_network_tools.yaml]: Yep, updated

File: .github/workflows/build_network_tools.yaml (hunk 3 above)
Comment (alexhumphreys) [reply .github/workflows/build_network_tools.yaml:60]: Might push latest for now, see if that burns us, if so I can kill it.

File: .github/workflows/build_network_tools.yaml (hunk 2 above)
Comment (alexhumphreys) [reply .github/workflows/build_network_tools.yaml:27]: Yeah I could just hardcode 'main' below...

**Pull request merged**